client/lib/titan/new-mailbox.js

```diff
@@ -125,6 +133,30 @@ const validateFullEmailAddress = ( { value, error }, allowEmpty = false ) => {
 	};
 };
 
+const validateAlternativeEmailAddress = ( { value, error }, domainName, allowEmpty = false ) => {
+	if ( ! error && value && domainName ) {
+		const parts = `${ value }`.split( '@' );
+		if ( parts.length > 1 && parts[ 1 ].toLowerCase() === domainName.toLowerCase() ) {
+			return {
+				value,
+				error: translate(
+					'This email address must be on a different domain than {{strong}}%(domain)s{{/strong}}. Please use a different email address.',
```

Comment:
The use of "on a different" domain feels weird to me, though I can't specify why!

It may be worth using something like the following:
```suggestion
					'This email address must use a different domain than {{strong}}%(domain)s{{/strong}}. Please use a different email address.',
```
I know this repeats "use" but I think that's better than using "be on". 😐